### DISCUSSION TITLE
How to set locale prop of ui.date?

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/nicegui#2793] [Q&A] [answered]
### Question

Quasar provides a [prop for custom locales](https://quasar.dev/vue-components/date#custom-ad-hoc-locale). How can this prop be set? The question came up [on Reddit](https://www.reddit.com/r/nicegui/comments/1brvg3t/change_uidate_props_locale/). This seems not to work:

```py
d = ui.date("2023-01-01").props("locale=\"{month:['A','B','C','D','E','F','G','H','I','J','K','L']}\"")
```

**Comment by falkoschindler** [comment, marked as answer]
You can either extend the local `_props` dictionary like this:
```py
d = ui.date('2023-01-01')
d._props['locale'] = {'months': ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L']}
```

<img width="305" alt="Screenshot 2024-03-31 at 11 06 49" src="https://github.com/zauberzeug/nicegui/assets/5767091/785d4c22-996b-44f3-b101-4ab5c00af4d9">

Or you use the `props()` method with a dynamic prop which will be evaluated on the client:
```py
ui.date('2023-01-01').props(''':locale="({months: ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L']})"''')
```

Note that we need to wrap the JavaScript object in parentheses. Otherwise the curly braces will be interpreted as a code block and not as an object literal.

And note that the key is called "months" and not "month".
